powerline_shell/segments/jobs.py

```diff
@@ -15,14 +29,19 @@ def run(self):
                 output_proc.communicate()[0].decode("utf-8").splitlines()[1:])
             self.num_jobs = output.count(os.getppid()) - 1
         else:
-            pppid_proc = subprocess.Popen(['ps', '-p', str(os.getppid()), '-oppid='],
-                                          stdout=subprocess.PIPE)
-            pppid = pppid_proc.communicate()[0].decode("utf-8").strip()
-            output_proc = subprocess.Popen(['ps', '-a', '-o', 'ppid'],
-                                           stdout=subprocess.PIPE)
+            pcmd = self.pidInfo(os.getppid(), "command")
+            
+            pppid = self.pidInfo(os.getppid(), "ppid")
+            
+            #fish runs commands in a sub process, so you have to walk up the
+            # tree one time to get back to where you were with bash or other
+            # such shells
+            if pcmd=="fish" or self.powerline.args.shell=="fish":
```

Comment:
I am guessing you are using `pcmd` AND `self.powerline.args.shell` just for exactness in case the user hasn't chosen `fish` as their shell? Or is there some other reason? I would like to just rely on `self.powerline.args.shell` if possible in order to avoid the additional subprocess.

Reply:
I found this problem specific to fish